extensions/cornerstone-dicom-rt/src/getSopClassHandlerModule.ts

```diff
@@ -72,22 +72,27 @@ function _getDisplaySetsFromSeries(
   displaySet.referencedImages = instance.ReferencedSeriesSequence.ReferencedInstanceSequence;
   displaySet.referencedSeriesInstanceUID = referencedSeries.SeriesInstanceUID;
 
-  displaySet.getReferenceDisplaySet = () => {
-    const { displaySetService } = servicesManager.services;
-    const referencedDisplaySets = displaySetService.getDisplaySetsForSeries(
-      displaySet.referencedSeriesInstanceUID
+  const { displaySetService } = servicesManager.services;
+  const referencedDisplaySets = displaySetService.getDisplaySetsForSeries(
+    displaySet.referencedSeriesInstanceUID
+  );
+
+  if (!referencedDisplaySets || referencedDisplaySets.length === 0) {
```

Comment:
Does this introduce a bug if one display set is already referenced, but a second one becomes available which would have been referenced had it been loaded?

Reply:
i don't think so since it is filtering based on `if (addedDisplaySet.SeriesInstanceUID === displaySet.referencedSeriesInstanceUID) {` ?